lib/engines/content_block_manager/app/services/content_block_manager/get_host_content_items.rb

```diff
@@ -4,18 +4,22 @@
 
 module ContentBlockManager
   class GetHostContentItems
-    attr_reader :content_id
+    attr_reader :content_id, :page, :order
 
-    def initialize(content_id:)
+    DEFAULT_ORDER = "-unique_pageviews".freeze
+
+    def initialize(content_id:, page: nil, order: nil)
       self.content_id = content_id
+      self.page = page
+      self.order = order || DEFAULT_ORDER
     end
 
-    def self.by_embedded_document(content_block_document:)
-      new(content_id: content_block_document.content_id).items
+    def self.by_embedded_document(content_block_document:, page: nil, order: nil)
+      new(content_id: content_block_document.content_id, page:, order:).items
```

Comment:
I wondered if you'd seen the `Kaminara` helper methods that Whitehall uses to paginate? https://github.com/alphagov/whitehall/blob/c88f75b785b92b5d045ac1e2d35c675ed98c414c/app/models/admin/edition_filter.rb#L26 
That means the results can just be passed to the `paginate` function in the `erb` file https://github.com/alphagov/whitehall/blob/adee19eb5cff5c1c83cb0b2ab46ece334539bd0b/lib/engines/content_block_manager/app/views/content_block_manager/content_block/documents/index.html.erb#L40 
it kind of hides away the logic/shape of data but is a bit cleaner, and you wouldn't need the extra data class that's been created?

Reply:
Yeah, we'd have to load all the results into memory then wouldn't we? (although, on reflection, that might have been easier in the first instance 😅 )